File: frontend/test-memory-api.js
// Test script for memory management API endpoints
import fetch from 'node-fetch';

const BASE_URL = 'http://localhost:5173';

// You'll need to get a valid token by logging in as pranav@idrinkcoffee.com
// For testing, you can use the browser's Network tab to copy the token after logging in
const AUTH_TOKEN = 'YOUR_AUTH_TOKEN_HERE';

async function testMemoryAPI() {
  console.log('Testing Memory Management API...\n');

  const headers = {
    'Authorization': `Bearer ${AUTH_TOKEN}`,
    'Content-Type': 'application/json'
  };

  try {
    // Test 1: Get all memories
    console.log('1. Testing GET /api/memory/all');
    const allMemoriesRes = await fetch(`${BASE_URL}/api/memory/all`, { headers });
    if (allMemoriesRes.ok) {
      const data = await allMemoriesRes.json();
      console.log(`✅ Success: Found ${data.total} total memories`);
      console.log(`   Users with memories: ${data.stats.byUser.length}`);
    } else {
      console.log(`❌ Failed: ${allMemoriesRes.status} ${allMemoriesRes.statusText}`);
    }

    // Test 2: Get memory statistics
    console.log('\n2. Testing GET /api/memory/stats');
    const statsRes = await fetch(`${BASE_URL}/api/memory/stats`, { headers });
    if (statsRes.ok) {
      const stats = await statsRes.json();
      console.log(`✅ Success: Stats retrieved`);
      console.log(`   Total memories: ${stats.total}`);
      stats.byUser.forEach(u => {
        console.log(`   User ${u.user_id}: ${u.count} memories`);
      });
    } else {
      console.log(`❌ Failed: ${statsRes.status} ${statsRes.statusText}`);
    }

    // Test 3: Search memories
    console.log('\n3. Testing GET /api/memory/search');
    const searchRes = await fetch(`${BASE_URL}/api/memory/search?q=coffee`, { headers });
    if (searchRes.ok) {
      const data = await searchRes.json();
      console.log(`✅ Success: Found ${data.count} memories matching "coffee"`);
      if (data.memories.length > 0) {
        console.log(`   First result: "${data.memories[0].memory || data.memories[0].content}"...`);
      }
    } else {
      console.log(`❌ Failed: ${searchRes.status} ${searchRes.statusText}`);
    }

    // Test 4: Get users with memories
    console.log('\n4. Testing GET /api/memory/users');
    const usersRes = await fetch(`${BASE_URL}/api/memory/users`, { headers });
    if (usersRes.ok) {
      const data = await usersRes.json();
      console.log(`✅ Success: Found ${data.users.length} users with memories`);
      data.users.forEach(u => {
        console.log(`   ${u.userId}: ${u.memoryCount} memories`);
      });
    } else {
      console.log(`❌ Failed: ${usersRes.status} ${usersRes.statusText}`);
    }

  } catch (error) {
    console.error('Error testing API:', error);
  }
}

console.log(`
To test the Memory Management API:

1. Start the EspressoBot server (npm run dev)
2. Login as pranav@idrinkcoffee.com
3. Open browser DevTools > Network tab
4. Look for any API request and copy the Authorization header token
5. Replace YOUR_AUTH_TOKEN_HERE with the actual token
6. Run: node test-memory-api.js

Note: The API will return 403 Forbidden if you're not logged in as pranav@idrinkcoffee.com
`);

// Uncomment to run tests:
// testMemoryAPI();

File: langgraph-backend/test_memory_api.py
#!/usr/bin/env python
"""Test the memory API endpoints"""
import asyncio
import aiohttp
import json

async def test_memory_api():
    base_url = "http://localhost:8000/api/memory"
    user_id = "1"
    
    async with aiohttp.ClientSession() as session:
        # Test dashboard endpoint
        print("\n1. Testing dashboard endpoint...")
        async with session.get(f"{base_url}/dashboard/{user_id}") as resp:
            print(f"   Status: {resp.status}")
            if resp.status == 200:
                data = await resp.json()
                print(f"   Stats: {data.get('stats', {})}")
                print(f"   Recent memories: {len(data.get('recent_memories', []))}")
            else:
                print(f"   Error: {await resp.text()}")
        
        # Test list endpoint
        print("\n2. Testing list endpoint...")
        async with session.get(f"{base_url}/list/{user_id}?limit=10") as resp:
            print(f"   Status: {resp.status}")
            if resp.status == 200:
                data = await resp.json()
                print(f"   Found {len(data)} memories")
                if data:
                    print(f"   First memory: {data[0].get('content', '')[:100]}...")
            else:
                print(f"   Error: {await resp.text()}")
        
        # Test search endpoint
        print("\n3. Testing search endpoint...")
        search_body = {
            "query": "coffee",
            "limit": 5,
            "similarity_threshold": 0.5
        }
        async with session.post(
            f"{base_url}/search/{user_id}", 
            json=search_body
        ) as resp:
            print(f"   Status: {resp.status}")
            if resp.status == 200:
                data = await resp.json()
                print(f"   Found {len(data)} matches")
            else:
                print(f"   Error: {await resp.text()}")
        
        print("\n✅ API test complete!")

if __name__ == "__main__":
    asyncio.run(test_memory_api())